### PR TITLE
[Chunk] Add "path" and "start_end_idx" at chunk return

### LINKED ISSUE #684: [Chunk] return the chunk filepath in `parsed_result` and what index that the chunk start and end

### PR DESCRIPTION
**Pull request by bwook00**
close #684

### CODE REVIEW

File: autorag/data/utils/util.py [hunk 1]
```diff
@@ -93,3 +94,11 @@ def get_param_combinations(modules: List[Dict]) -> Tuple[List[Callable], List[Di
...
 	module_list, combination_list = explode(module_callable_list, combinations)
 	return module_list, combination_list
+
+
+def gen_start_end_idx(contents: List[str]) -> List[Tuple[int, int]]:
+	lengths = [len(content) for content in contents]
+	start_indices = list(accumulate([0] + lengths[:-1]))
+	return [
+		(start, start + length - 1) for start, length in zip(start_indices, lengths)
+	]
```
Comment (vkehfdl1) [inline autorag/data/utils/util.py:104]: I think this will not work when chunk_overlap set...

File: autorag/data/chunk/llama_index_chunk.py [hunk 2]
```diff
@@ -14,42 +19,66 @@ def llama_index_chunk(
...
+
+	doc_id, contents, path, start_end_idx, metadata = (
+		list(chain.from_iterable(item)) for item in zip(*results)
+	)
+
+	return list(doc_id), list(contents), list(path), list(start_end_idx), list(metadata)
 
-	# chunk documents
-	chunk_results = chunker.get_nodes_from_documents(documents=documents)
+
```
Comment (vkehfdl1) [inline autorag/data/chunk/llama_index_chunk.py:53]: Why did you use async at here? 
There are no await in the code, so async will be useless.
It only add overhead to the event loop

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

File: tests/autorag/data/chunk/test_chunk_base.py [hunk 3]
```diff
@@ -59,12 +59,38 @@
...
 }
 
+expect_token_path = [
+	base_metadata[0]["path"],
+	base_metadata[0]["path"],
+	base_metadata[1]["path"],
+	base_metadata[1]["path"],
+]
+
+expect_token_idx = [(0, 143), (144, 169), (0, 119), (120, 166)]
```
[the remark this answers is not in the document]
Comment (vkehfdl1) [reply tests/autorag/data/chunk/test_chunk_base.py:69]: I mean add test code for overlapping chunk

**Comment by vkehfdl1** [comment]
@bwook00 And will you allow semantic chunking?? It will not work properly yet.

File: autorag/data/utils/util.py [hunk 4]
```diff
@@ -93,3 +93,9 @@ def get_param_combinations(modules: List[Dict]) -> Tuple[List[Callable], List[Di
 	combinations = list(map(make_combinations, module_params_list))
 	module_list, combination_list = explode(module_callable_list, combinations)
 	return module_list, combination_list
+
+
+def get_start_end_idx(original_text: str, search_str: str) -> Tuple[int, int]:
+	start_idx = original_text.find(search_str)
+	end_idx = start_idx + len(search_str)
+	return start_idx, end_idx
```
[the remark this answers is not in the document]
Comment (bwook00) [reply autorag/data/utils/util.py]: 1. Oh that is my mistake 😂
2. make issue #689 !!

**Review by vkehfdl1 — APPROVED** [review]
LGTM

**Pull request merged**